docs/_docs/02_features/keep-your-remote-state-configuration-dry.md

```diff
@@ -135,6 +137,58 @@ The child `.hcl` file’s `terraform` settings will be merged into the parent fi
 
 Other settings in the child `.hcl` file override the respective settings in the parent.
 
+
+### Using the generate property to generate terraform code for managing remote state
+
+While the default way terragrunt manages remote state is through `terraform init` with `-backend-config`, you can also
+use the `generate` property to configure terragrunt to geneerate a `.tf` file in the terraform working directory with the
+backend configuration.
+
+The `generate` property is an object that accepts two parameters:
+
+- `path`: The path where the generated file should be written. If a relative path, it'll be relative to the Terragrunt
+  working dir (where the terraform code lives).
+- `if_exists`: What to do if a file already exists at `path`. Valid values are: `overwrite` (overwrite the existing
+  file), `skip` (skip code generation and leave the existing file as-is), `error` (exit with an error).
+
+For example, here is a version of the root `remote_state` configuration with the `generate` property:
+
+```hcl
+remote_state {
+  backend = "s3"
+  generate = {
```

Comment:
Does this code generation happen every time you run Terragrunt? Or only during Auto Init?

Reply:
Every time. See the commentary right below this, where it says:

> With this configuration, `terragrunt` will generate a new file `backend.tf` in the working directory before it calls out to any `terraform` command with the following contents: